plugins/woocommerce/tests/e2e-pw/tests/merchant/order-edit.spec.js

```diff
@@ -86,6 +86,77 @@ test.describe( 'Edit order', () => {
 			'2018-12-14'
 		);
 	} );
+
+	test( 'can load billing details', async ( { page, baseURL } ) => {
+		let customerId = 0;
+
+		const api = new wcApi( {
+			url: baseURL,
+			consumerKey: process.env.CONSUMER_KEY,
+			consumerSecret: process.env.CONSUMER_SECRET,
+			version: 'wc/v3',
+		} );
+
+		// Debug note: this request will fail with HTTP status 400 if the test is re-run without first resetting state.
+		await api
+			.post( 'customers', {
+				email: 'archie123@email.addr',
+				first_name: 'Archie',
+				last_name: 'Greenback',
+				username: 'big.archie',
+				billing: {
+					first_name: 'Archibald',
+					last_name: 'Greenback',
+					company: 'Automattic',
+					country: 'US',
+					address_1: 'address1',
+					address_2: 'address2',
+					city: 'San Francisco',
+					state: 'CA',
+					postcode: '94107',
+					phone: '123456789',
+					email: 'archie123@email.addr',
+				}
+			} )
+			.then( ( response ) => {
+				customerId = response.data.id;
+			} );
+
+		// Open our test order and select the customer we just created.
+		await page.goto( `wp-admin/post.php?post=${orderId}&action=edit` );
+		await page.evaluate(
+			( customerId ) => document.getElementById( 'customer_user' ).innerHTML = `<option value=${customerId} selected></option>`,
+			customerId
+		);
+
+		// Simulate the ajax `woocommerce_get_customer_details` call normally done inside meta-boxes-order.js.
+		const response = await page.evaluate(
+			async ( customerId ) => {
+				const simulateCustomerDetailsCall = new Promise( ( resolve ) => {
+					jQuery.ajax( {
```

Comment:
I don't think we need to resort to jQuery for this.  Playwright has some [built-in handling for ajax/network requests](https://playwright.dev/docs/network#network-events).  

It's cool if this is working, but we should probably look at refactoring it into a more Playwright-y way in the future (and we can probably also use that customer-role user as mentioned above)

Reply:
Yep, I was really just mimicking the way `wc_meta_boxes_order.load_billing()` works, but you're right: jQuery isn't strictly necessary.

My first iteration tried to be a little more Playwrighty and clicked the actual *Load billing address* link (though, under the hood, this would still invoke basically the same jQuery-based ajax call), then used the network methods you linked to, to try and observe/capture the response—but I was unable to get it functioning, and went with this approach to expedite things.

<div align="center"><img width="650" alt="load-billing" src="https://github.com/woocommerce/woocommerce/assets/3594411/9362e088-cf19-4041-b330-fa1ee829d77d"></div>

Definitely would vote for refactoring in the future vs now, as it took a surprising amount of time to get this in place (probably because of a lack of Playwright knowledge).

Reply:
I'm always a fan of "make it work" before all else!  This sounds like a fine approach to me.